### PR TITLE
fix: correct min max time

### PR DESCRIPTION
**Pull request by Gi729sdF5hdf**

### CODE REVIEW

File: rule/time.go [hunk 1]
```diff
@@ -93,23 +95,23 @@ func (t Time) ValidateValue(value any) error {
 
 	result := NewResult()
 
-	if t.min != nil && vt.Before(*t.min) {
+	if t.min != nil && vt.Before(t.min()) {
```
Comment (raoptimus) [inline rule/time.go:98]: need to define max min  before using

File: rule/time.go [hunk 2]
```diff
@@ -6,14 +6,16 @@ import (
 	"github.com/raoptimus/validator.go/ctype"
 )
 
+type FuncTime func() time.Time
```
Comment (raoptimus) [inline rule/time.go:9]: TimeFunc will be better

Comment (Gi729sdF5hdf) [reply rule/time.go]: done

File: rule/time.go (hunk 1 above)
Comment (Gi729sdF5hdf) [reply rule/time.go]: done

File: rule/time.go [hunk 3]
```diff
@@ -93,23 +95,25 @@ func (t Time) ValidateValue(value any) error {
 
 	result := NewResult()
 
-	if t.min != nil && vt.Before(*t.min) {
+	minTime := t.min()
+	if t.min != nil && vt.Before(minTime) {
```
Comment (raoptimus) [inline rule/time.go:99]: do you see anything suspicious?

Comment (Gi729sdF5hdf) [reply rule/time.go]: done

**Pull request merged**